<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- Camera topic and weights, config and classes files -->
    <arg name="image_topic"                 default="/usb_cam/image_raw"/>
    <arg name="weights"                     default="yolov5s.pt"/>
    <arg name="imgsz"                       default="640"/>
    <arg name="conf-thres"                  default="0.25"/>
    <arg name="iou-thres"                   default="0.45"/>
    <arg name="max-det" default="1000"/>
    <arg name="device" default=""/>

    <!-- Published topics -->
    <arg name="publish_image"	            default="true"/>
    <arg name="detected_objects_topic"      default="yolov5/bounding_boxes"/>
    <arg name="detections_image_topic"      default="yolov5/image_output"/>
    <arg name="bbox_line_thickness" default="3"/>
    <!-- Node -->
    <node name="detector_manager_node" pkg="yolov5_pytorch_ros" type="detect.py" output="screen" respawn="true">
        <param name="image_topic" value="$(arg image_topic)"/>
        <param name="weights" value="$(arg weights)"/>
        <param name="imgsz" value="$(arg imgsz)"/>
        <param name="conf-thres" value="$(arg conf-thres)"/>
        <param name="iou-thres" value="$(arg iou-thres)"/>
        <param name="max-det" value="$(arg max-det)"/>
        <param name="device" value="$(arg device)"/>
        <param name="publish_image" value="$(arg publish_image)"/>
        <param name="detected_objects_topic" value="$(arg detected_objects_topic)"/>
        <param name="detections_image_topic" value="$(arg detections_image_topic)"/>
        <param name="bbox_line_thickness" value="$(arg bbox_line_thickness)"/>
    </node>
</launch>
